src/components/refactor/dialogs/two-windings-transformer-creation/tap-changer-pane/tap-changer-pane-taps.js

```diff
@@ -0,0 +1,368 @@
+    const [openAddRowsDialog, setOpenAddRowsDialog] = useState(false);
+
+    function handleAddRowsButton() {
+        // triggering validation on low tap position before generating rows (the field is required)
+        trigger(`${tapChanger}.${LOW_TAP_POSITION}`).then((result) => {
+            // if the trigger returns false, it means the field validation didn't pass -> we don't generate rows
+            // the user will see the low tap field in red
+            if (result) {
+                setOpenAddRowsDialog(true);
+            }
+        });
+    }
+
+    function addNewRows(numberOfRows) {
+        const currentLowTapPosition = getValues(
+            `${tapChanger}.${LOW_TAP_POSITION}`
+        );
+        const currentTapRows = getValues(`${tapChanger}.${STEPS}`);
+
+        // checking if not exceeding 100 steps
+        if (currentTapRows.length + numberOfRows > MAX_TAP_NUMBER) {
+            setError(`${tapChanger}.${STEPS}`, {
+                type: 'custom',
+                message: {
+                    id: 'TapPositionValueError',
+                    value: MAX_TAP_NUMBER,
+                },
+            });
+            return;
+        }
+
+        let currentHighestTap;
+        if (currentTapRows.length === 0) {
+            currentHighestTap = currentLowTapPosition - 1; // we do + 1 right after
+        } else {
+            currentHighestTap =
+                currentTapRows[currentTapRows.length - 1][STEPS_TAP];
+        }
+
+        const tapRowsToAdd = [];
+        for (let i = 0; i < numberOfRows; i++) {
+            // we directly use currentHighestTap + 1 as first tap value
+            currentHighestTap++;
+            // we remove STEPS_TAP from the columns with slice
+            const newRow = columnsDefinition.slice(1).reduce(
+                (accumulator, currentValue) => ({
+                    ...accumulator,
+                    [currentValue.dataKey]: currentValue.initialValue,
+                }),
+                { [SELECTED]: false, [STEPS_TAP]: currentHighestTap }
+            );
+            tapRowsToAdd.push(newRow);
+        }
+
+        // note : an id prop is automatically added in each row
+        append(tapRowsToAdd);
+        setValue(`${tapChanger}.${HIGH_TAP_POSITION}`, currentHighestTap);
+    }
+
+    function handleCloseAddRowsDialog() {
+        setOpenAddRowsDialog(false);
+    }
+
+    function deleteSelectedRows() {
+        const currentTapRows = getValues(`${tapChanger}.${STEPS}`);
+
+        let rowsToDelete = [];
+        for (let i = 0; i < currentTapRows.length; i++) {
+            if (currentTapRows[i][SELECTED]) {
+                rowsToDelete.push(i);
+            }
+        }
+
+        remove(rowsToDelete);
+
+        const currentHighTapPosition = getValues(
+            `${tapChanger}.${HIGH_TAP_POSITION}`
+        );
+
+        let newHighTapPosition;
+        if (currentTapRows.length === rowsToDelete.length) {
+            newHighTapPosition = null;
```

Comment:
IS there a problem when setting HIGH_TAP_POSITION to 0 ?

Reply:
Yes, high tap position is not the number of rows but the highest tap number of every rows. If no rows, no high tap position IMO.